dgv/simplifions/task_functions.py

```diff
@@ -11,44 +12,63 @@
 
 dgv_headers = {"X-API-KEY": DATAGOUV_SECRET_API_KEY}
 
+GRIST_DOC_ID = "c5pt7QVcKWWe"
+MAIN_TABLE_ID = "SIMPLIFIONS_cas_usages"
+SUBDATA_TABLE_IDS = {
+    # "Solutions_publiques_recommandees": "SIMPLIFIONS_produitspublics",
+    "reco_solutions": "SIMPLIFIONS_reco_solutions_cas_usages",
+    "usagers": "TYPE_usagers",
+    "A_destination_de_": "TYPE_fournisseur_service",
+}
 
 def get_and_format_grist_data(ti):
-    doc_id = "c5pt7QVcKWWe"
-    table_id = "SIMPLIFIONS_cas_usages"
+    rows = request_grist_table(MAIN_TABLE_ID)
+    
+    grist_topics = {
+        row["slug"]: get_and_format_grist_subdata(row)
+        for row in rows
+        if row["slug"]
+    }
+    ti.xcom_push(key="grist_topics", value=grist_topics)
+
+def get_and_format_grist_subdata(row):
+    formatted_row = {
+        key: handle_subdata_list(key, row[key])
+        for key in row.keys()
+    }
+    return formatted_row
+    
+def handle_subdata_list(key, value):
+    if key in SUBDATA_TABLE_IDS.keys():
+        if isinstance(value, list) and value[0] == "L":
+            filter = json.dumps({ "id": value[1:] })
+            print(filter)
+            return request_grist_table(SUBDATA_TABLE_IDS[key], filter=filter)
+        else:
+            return value
+    else:
+        return value

```

Comment:
```suggestion
    if key in SUBDATA_TABLE_IDS.keys() and isinstance(value, list) and value[0] == "L":
        filter = json.dumps({ "id": value[1:] })
        print(filter)
        return request_grist_table(SUBDATA_TABLE_IDS[key], filter=filter)
    else:
        return value
```
but I don't get why we want to retrieve the data once more instead of just popping the `"L"`

Reply:
Je suis pas sûr d'avoir compris le commentaire, mais dans le doute j'ai quand même essayé de mieux faire : 

J'ai séparé le cleanup des "L" et la récupération des subdata.